commitizen/git.py

```diff
@@ -168,7 +168,22 @@ def get_tags(dateformat: str = "%Y-%m-%d") -> List[GitTag]:
 
 def tag_exist(tag: str) -> bool:
     c = cmd.run(f"git tag --list {tag}")
-    return tag in c.out
+    if tag in c.out:
+        return True
+
+    # In shallow clones (e.g. set up by Gitlab CI), the previous release tag might not
```

Comment:
I'm good with this functionality. But one of the concerns I have is whether this is an unexpected action from the user point of view. I'm thinking of using a flag.

Reply:
Agreed, this is more CI configuration, is up to the user, I'm fine with a warning and an explanation with the next steps to the user. But the user should ensure the environment (git) is providing commitizen what's needed, not the other way around.

Reply:
Thanks for the reviews!

I agree with your concerns that auto-fetching missing tags is unexpected from the user point of view. Actually it would be cleanest to me if `git tag --list {tag}` had a flag to do this fetching for shallow clones automatically. But without auto-fetching missing tags in shallow clones, the only option would be to do a full clone, which can be quite slow for large/old repositories.

I believe both Gitlab CI and Github Actions use shallow clones by default. Because of that, Commitizen should ideally handle shallow clones out-of-the-box without setting an extra flag. Do you have examples which workflows could break if we fetch in shallow clones by default? I'm mostly worried about connection issues/the CI not setting up the Git remote for later fetching.

If you keep not being convinced, should we call the flag `--auto-fetch` and only add it for the `cz bump` command for now?

Reply:
Hi @robertschweizer Sorry for late reply. I like the idea of the flag `--auto-fetch`. @woile What do you think?

Reply:
Yes, a flag and a setting. We can prompt the user in the init command